### ISSUE TITLE
Name Learning

### ISSUE THREAD

**Issue by efvaldez** [RasaHQ/rasa_nlu#653]
Hi All, can you use both ner_crf and ner_spacy as entity extractor?
I plan to create a chatbot for registration using RASA NLU , I plan to get the name ( First Name would be fine for now) and email address.
Using ner_crf , i placed around 50 first names (some starts with capital letters while some do not) it does not trigger my entity "first_name" if the name is not capitalized and not in my training data. If the names is not in my training data but starts with a capital letter, RASA NLU mostly can trigger my "first_name" entity. Any tips on name learning? I am okay with first name for now. Thanks All!

<img width="959" alt="test1" src="https://user-images.githubusercontent.com/33081604/32043759-3d00e0ac-ba6e-11e7-8477-1e3d96140980.png">
<img width="954" alt="test2" src="https://user-images.githubusercontent.com/33081604/32043762-3ddd637e-ba6e-11e7-8b53-34840cfafd1c.png">

In these images, I did not place "donald" or "Donald" in my training data.

**Comment by tmbo** [comment]
The crf component is case sensitive by default. You can change that though by modifying the features the CRF is looking at:
```
  "ner_crf": {
    "BILOU_flag": true,
    "features": [
      ["low", "title", "upper", "pos", "pos2"],
      ["bias", "low", "word3", "word2", "upper", "title", "digit", "pos", "pos2", "pattern"],
      ["low", "title", "upper", "pos", "pos2"]],
    "max_iterations": 50,
    "L1_c": 1,
    "L2_c": 1e-3
  }
```
You need to remove the `"low"` from the feature lists and add the whole `"ner_crf"` config to your models configuration.

**Comment by amn41** [comment]
actually `low` is the whole word just lowercased. Removing this means the CRF doesn't have access to the identity of the word (which may or may not be desired)

Full set of features is here : https://sklearn-crfsuite.readthedocs.io/en/latest/tutorial.html#features

I think you want to remove the `title` feature

**Comment by tmbo** [comment]
Ah right, sorry for that.

**Comment by wrathagom** [comment]
@efvaldez did the above help answer your questions?

to answer your other one, yes you can use ner_spacy with ner_crf just add it to your pipeline:

```["nlp_spacy", "tokenizer_spacy", "intent_entity_featurizer_regex", "intent_featurizer_spacy", "ner_crf", "ner_spacy", "ner_synonyms",  "intent_classifier_sklearn"]
```

**Comment by intuitmachine** [comment]
Hello Sir @tmbo @wrathagom @amn41  sorry for the late reply, thank you so much for the answers, I will try these right now.  The chatbot I am creating is for registration (name,email etc.) for now. Would it be better if I configure ner_crf to accept lower case names instead of using nlp_spacy's Person Entity for the name?

Lastly, around how many training data should I use for ner_crf to recognize first names in lower or upper case? Thanks

**Comment by efvaldez1** [comment]
Hello Sir @tmbo @wrathagom @amn41 sorry for the late reply, thank you so much for the answers, I will try these right now. The chatbot I am creating is for registration (name,email etc.) for now. Would it be better if I configure ner_crf to accept lower case names instead of using nlp_spacy's Person Entity for the name?

Lastly, around how many training data should I use for ner_crf to recognize first names in lower or upper case? Thanks

**Comment by efvaldez1** [comment]
Hi All, I want to create a chat bot for registration, for now I am okay with name(first name is okay for now) and email address. I removed the title feature in config_defaults.json  
![image](https://user-images.githubusercontent.com/8022105/32161146-7a5eadca-bd8f-11e7-9c2e-7b684ea18d2f.png)

However I find it weird that now , when I try to input "vincent" and "brian which are not in my training data and then I tried "eduardo" which is in my training , the entity is first_name which is correct but the intent is get_email, which should be get_name.

![image](https://user-images.githubusercontent.com/8022105/32161333-1de1ae5c-bd90-11e7-9e04-3c429f1fcb7a.png)

![image](https://user-images.githubusercontent.com/8022105/32161337-23d1e908-bd90-11e7-9fd8-446c58ac5a1d.png)


here is my training data 

[testjson.txt](https://github.com/RasaHQ/rasa_nlu/files/1426483/testjson.txt)

**Comment by efvaldez1** [comment]
Hi All, I tried adding regex features, but I get "ValueError": Unknown Training format, even thou I copied and paste this:
"rasa_nlu_data": {
        "regex_features": [
            {
                "name": "zipcode",
                "pattern": "[0-9]{5}"
            },
            {
                "name": "greet",
                "pattern": "hey[^\s]*"
            },
        ]

From the tutorial http://rasa-nlu.readthedocs.io/en/latest/dataformat.html?highlight=regex_features#regular-expression-features
But if I only used the zipcode,it works. Why so? Thanks

**Comment by wrathagom** [comment]
Sorry I don't see any get_email intents in the training data?

Also, I am pretty sure duckling can do e-mail: https://rasa-nlu.readthedocs.io/en/latest/pipeline.html#ner-duckling

**Comment by wrathagom** [comment]
Closing @efvaldez let us know if you still had more questions.

**Comment by efvaldez1** [comment]
Good day, I am very sorry for the late reply. May I ask with this training data 
[textjson.txt](https://github.com/RasaHQ/rasa_nlu/files/1466439/textjson.txt)

and with this as my config_defaults.json
[config_defaults_json.txt](https://github.com/RasaHQ/rasa_nlu/files/1466460/config_defaults_json.txt)

My app could still not get all email and first_name intents:
![image](https://user-images.githubusercontent.com/8022105/32721571-deb1086e-c8a1-11e7-81ad-5397647dcd68.png)

The conversation goes as:
Me: hi
bot: what is your name?
Me: ludovic is my name.

Then it goes index zero for it does not have a first_name intent in the response. I plan to create a simple registration chatbot that gets name (first name for now) and email. When I test my bot using other names and emails and if it triggers get_name and get_email intent and with first_name and email entities , it gives me 80%+ accuracy. However when I test it with a name like "ludovic" from wiki leaks , it is not accurate. What can I do? Thanks

**Comment by ujja** [comment]
Hello!

I want achieve the same. I think if user enters just the name
Me: hi
bot: what is your name (just name please)?
Me: ludovic

http://rasa.com/docs/core/slotfilling/ freetextinput can help or,
https://github.com/mrbot-ai/rasa-addons swap intents

**Comment by GraphGrailAi** [comment]
I need a better names understanding too, should i use giant vectors to achieve this?
